cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js

```diff
@@ -117,20 +123,38 @@ export default class FilterDirective extends Component {
 
     let condition = DIRECTIVES_MAP[this.state.rowFilter][this.state.selectedCondition];
 
+    let configuration;
+
     switch (this.state.selectedCondition) {
       case 'EMPTY':
         directive = `${condition} ${column} ^\\s*$`;
         break;
       case 'TEXTCONTAINS':
+        if (this.state.ignoreCase) {
+          textValue = `(?i)${textValue}`;
+        }
         directive = `${condition} ${column} .*${textValue}.*`;
         break;
       case 'TEXTSTARTSWITH':
-        directive = `${condition} ${column} =^ "${textValue}"`;
+        configuration = `"${textValue}"`;
+        if (this.state.ignoreCase) {
+          column = `${column}.toLowerCase()`;
```

Comment:
If the user entered column name is lowercase and if the column name is actually in the uppercase + user checks ignore case checkbox, would the backend know if it should ignore case here?